src/http-api.ts

```diff
@@ -82,14 +93,21 @@ export class HttpApi {
       const requestDataStream = (parseInt(contentLength) > 0 || transferEncoding !== undefined) ? req : undefined;
 
       const requestContext: RequestContext = { dwn: this.dwn, transport: 'http', dataStream: requestDataStream };
-      const { jsonRpcResponse, dataStream: responseDataStream } = await jsonRpcApi.handle(dwnRequest, requestContext as RequestContext);
+      const { jsonRpcResponse, dataStream: responseDataStream } = await jsonRpcApi.handle(dwnRpcRequest, requestContext as RequestContext);
 
       // If the handler catches a thrown exception and returns a JSON RPC InternalError, return the equivalent
       // HTTP 500 Internal Server Error with the response.
       if (jsonRpcResponse.error) {
+        this.requestCounter.inc({ method: dwnRpcRequest.method, error: 1 });
         return res.status(500).json(jsonRpcResponse);
       }
 
+      if(jsonRpcResponse.result && jsonRpcResponse.result.reply && jsonRpcResponse.result.reply.status && jsonRpcResponse.result.reply.status.code !== undefined) {
```

Comment:
don't we support optional chaining?
```suggestion
      if (jsonRpcResponse?.result?.reply?.status?.code !== undefined) {
```

Reply:
right, thanks. Still remembering all the wacky javascriptisms